mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala

```diff
@@ -35,6 +35,8 @@ import org.apache.spark.sql.types._
  */
 @Experimental
 @Since("2.3.0")
+@deprecated("use `spark.read.format(\"image\").load(path)` and this `ImageSchema` will be " +
```

Comment:
There are other methods defined under `ImageSchema` that are not covered by the image data source. So we shall only deprecate `readImages` and leave other public methods as experimental. Same for Python.